src/trade/createSelectorForImmutable.js

```diff
@@ -0,0 +1,7 @@
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import Immutable from 'immutable';
+
+function createSelectorForImmutable(...args) {
+    return createSelectorCreator(defaultMemoize, (a, b) => a === b || Immutable.is(a, b))(...args);
+}
+export default createSelectorForImmutable;
```

Comment:
@borisyankov 
This is a selector that compare via deep equality, you can ignore other code